### PR TITLE
change name of probe to is_ready within ck::future

### PR DESCRIPTION
**Pull request by Wingpad**
ensures consistency with C++ futures. this feature is likely not used yet, so this is the optimal time to change it

### CODE REVIEW

**Review by evan-charmworks — APPROVED** [review]

**Comment by evan-charmworks** [comment]
@matthiasdiener It looks like the Spack CI cannot handle branches from external repositories.

**Comment by matthiasdiener** [comment]
> @matthiasdiener It looks like the Spack CI cannot handle branches from external repositories.

Yeah, that's true. Feel free to merge this with the test failing for now, passing the test is not required for a merge.

**Comment by matthiasdiener** [comment]
For pull requests from forks, we will need to merge #3177 first.

**Review by stwhite91 — APPROVED** [review]

File: src/ck-core/ckfutures.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ namespace ck {
       CkSendToFuture(handle_, msg);
     }
 
-    bool probe() { return CkProbeFuture(handle_); }
+    bool is_ready() { return CkProbeFuture(handle_); }
```
Comment (NK-Nikunj) [inline src/ck-core/ckfutures.h:87]: In reference to Prof. Kale's comment on the proposed change, can we keep both `is_ready` and `probe` as synonyms to keep it similar to both C++ future's member function `is_ready`, and MPI's `probe` function?

**Review by NK-Nikunj — CHANGES_REQUESTED** [review]

File: src/ck-core/ckfutures.h (hunk 1 above)
Comment (Wingpad) [reply src/ck-core/ckfutures.h:87]: Probe is retained within the C-like interface, which can serve as a frame-of-reference for MPI users. For this reason, and the fact that they are documented side-by-side, I do not think an alias is necessary within the C++ interface.

Comment (NK-Nikunj) [reply src/ck-core/ckfutures.h:87]: Thanks for the clarification. LGTM thanks!

**Review by NK-Nikunj — APPROVED** [review]

**Pull request merged**